pkg/imgpkg/bundle/images_lock.go

```diff
@@ -79,6 +111,23 @@ func (o *ImagesLock) LocalizeImagesLock() (lockconfig.ImagesLock, bool, error) {
 	return imagesLock, false, nil
 }
 
+func (o ImagesLock) LocationPrunedImageRefs() ([]lockconfig.ImageRef, error) {
+	var imageRefs []lockconfig.ImageRef
+	for _, imgRef := range o.imagesLock.Images {
+		newImgRef := imgRef
```

Comment:
locations inside imageRef wont be copied as a separate slice. we should probably add DeepCopy() to ImageRef and use it here.